aggregation/src/main/java/fk/prof/aggregation/serialize/Serializer.java

```diff
@@ -0,0 +1,47 @@
+package fk.prof.aggregation.serialize;
+
+import com.google.protobuf.Message;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.zip.Adler32;
+
+/**
+ * Base class for serializers.
+ * Provided helper methods to serialize Integer and protobuf message.
+ * Protobuf message will be serialized in the format [size][message][checksum].
+ * @author gaurav.ashok
+ */
+public abstract class Serializer<T> {
+
+    public abstract void serialize(T object, OutputStream os) throws IOException;
+
+    public static void writeInt32(int value, OutputStream os) throws IOException {
+        byte[] bytes = {(byte)(value >> 24), (byte)(value >> 16), (byte)(value >> 8), (byte)value};
```

Comment:
I wanted the order of bytes to be transparent in code. Thats why i didn't used any other lib's method.